src/audio/kpb.c

```diff
@@ -763,9 +829,15 @@ static int kpb_prepare(struct comp_dev *dev)
 		kpb->clients[i].r_ptr = NULL;
 	}
 
+#if CONFIG_AMS
+	/* AMS Register KPB for notification */
+	ret = kpb_register_ams_consumer(dev);
+#else
 	/* Register KPB for notification */
 	ret = notifier_register(dev, NULL, NOTIFIER_ID_KPB_CLIENT_EVT,
 				kpb_event_handler, 0);
+#endif /* CONFIG_AMS */
```

Comment:
presumably you could make `kpb_event_handler()` depend on `!CONFIG_AMS`. In fact it probably should be done to avoid a compiler warning about an unused static function

Reply:
Makes sense. Fixed.